core/templates/dev/head/services/RteHelperService.js

```diff
@@ -181,9 +182,29 @@ oppia.factory('RteHelperService', [
           });
         });
 
-        text = elt.html();
-        text = $filter('removeExtraLines')(text);
-        return text;
+        var textElt = elt[0].childNodes;
+        for (var i = textElt.length - 1; i > 0; i--) {
+          if (textElt[i].localName !== 'p') {
+            break;
+          }
+          if (textElt[i].childNodes.length === 0) {
```

Comment:
I don't think you need lines 190-193, right?

Reply:
Actually this is for the time when there is something like ```<p></p>```. So for this case I want to remove the tag right away and don't want to execute further. But if I remove these lines (190-193), the code will execute the for loop and give console error ```Cannot read property length of undefined```

Reply:
I don't understand. Wouldn't j just initialize to 0 in that case?

Reply:
Actually not, the value of j is ```textElt[i].childNodes.length```, for which it'll try to find childnodes of j, but will be undefined and accessing property of undefined will give a console error.

Reply:
I still don't understand. This condition only triggers when textElt[i].childNodes.length = 0. So, if you say `j = textElt[i].childNodes.length` wouldn't j then be assigned the value 0?